zntrack/core/functions/decorator.py

```diff
@@ -0,0 +1,218 @@
+import dataclasses
+import logging
+import pathlib
+import subprocess
+import typing
+
+import dot4dict
+
+from zntrack.core.dvcgraph import DVCRunOptions
+from zntrack.utils import file_io, utils
+
+str_or_path = typing.Union[str, pathlib.Path]
+
+log = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class NodeConfig:
+    """DataClass to contain the arguments passed by the user
+
+    Attributes
+    ----------
+    All dvc attributes but connected by "_" instead of "-"
+
+    """
+
+    deps: typing.Union[str_or_path, typing.List[str_or_path]] = None
+    outs: typing.Union[str_or_path, typing.List[str_or_path]] = None
+    outs_no_cache: typing.Union[str_or_path, typing.List[str_or_path]] = None
+    outs_persist: typing.Union[str_or_path, typing.List[str_or_path]] = None
+    outs_persist_no_cache: typing.Union[str_or_path, typing.List[str_or_path]] = None
+    metrics: typing.Union[str_or_path, typing.List[str_or_path]] = None
+    metrics_no_cache: typing.Union[str_or_path, typing.List[str_or_path]] = None
+    plots: typing.Union[str_or_path, typing.List[str_or_path]] = None
+    plots_no_cache: typing.Union[str_or_path, typing.List[str_or_path]] = None
+
+    params: typing.Union[dot4dict.dotdict, dict] = dataclasses.field(default_factory=dict)
+
+    def write_dvc_command(self, node_name: str) -> list:
+        """Collect dvc commands
+
+        Parameters
+        ----------
+        node_name:str
+            name of the node, usually func.__name__
+
+        Returns
+        -------
+        list: a list of all options like
+        ["--outs", "outs.txt", "--params", "params.yaml:<node_name>", ...]
+        handling lists of files as well the parameters
+
+        """
+        script = []
+        if self.params is not None:
+            if len(self.params) > 0:
+                script += ["--params", f"params.yaml:{node_name}"]
+        for field in self.__dataclass_fields__:
+            if field == "params":
+                continue
+            if isinstance(getattr(self, field), (list, tuple)):
+                for element in getattr(self, field):
+                    script += [f"--{field.replace('_', '-')}", str(element)]
+            elif getattr(self, field) is not None:
+                script += [f"--{field.replace('_', '-')}", str(getattr(self, field))]
+
+        return script
+
+
+any_or_nodeconfig = typing.Union[typing.Any, NodeConfig]
```

Comment:
not sure how much is gained here, given the standardization of the Union I may just leave the full thing when it is needed.

Reply:
It helps with typing, because it autocompletes for `NodeConfig`